subsys/net/lib/wifi_credentials/wifi_credentials_backend_settings.c

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <stdlib.h>
+#include <zephyr/settings/settings.h>
+
+#include "wifi_credentials_internal.h"
+
+LOG_MODULE_REGISTER(wifi_credentials_backend, CONFIG_WIFI_CREDENTIALS_LOG_LEVEL);
+
+extern struct k_mutex wifi_credentials_mutex;
+
+
+BUILD_ASSERT(ENTRY_MAX_LEN <= SETTINGS_MAX_VAL_LEN);
+
+/**
+ * @brief WiFi credentials settings handler
+ * @note This callback function is used to initialize the SSID cache.
+ *
+ * @param key credential index
+ * @param len length of entry
+ * @param read_cb read callback
+ * @param cb_arg argument for read callback
+ * @return 0 on success, otherwise a negative error code
+ */
+static int settings_set(const char *key, size_t len, settings_read_cb read_cb, void *cb_arg)
+{
+	int ret = 0;
+
+	/* key validation */
+	if (!key) {
+		LOG_ERR("Settings error: no key");
+		return -EINVAL;
+	}
+
+	k_mutex_lock(&wifi_credentials_mutex, K_FOREVER);
+
+	int idx = atoi(key);
+
+	if ((idx == 0 && strcmp(key, "0") != 0) ||
+	     idx >= CONFIG_WIFI_CREDENTIALS_MAX_ENTRIES) {
+		LOG_ERR("Settings error: index too large");
+		ret = -EINVAL;
+		goto exit;
+	}
+
+	if (len < sizeof(struct wifi_credentials_header)) {
+		LOG_ERR("Settings error: invalid settings length");
+		ret = -EINVAL;
+		goto exit;
+	}
+
+	uint8_t buf[ENTRY_MAX_LEN];
+	size_t length_read = read_cb(cb_arg, buf, ARRAY_SIZE(buf));
+
+	/* value validation */
+	if (length_read < len) {
+		LOG_ERR("Settings error: entry incomplete");
+		ret = -ENODATA;
+		goto exit;
+	}
+
+	wifi_credentials_cache_ssid(idx, (struct wifi_credentials_header *) buf);
+
+exit:
+	k_mutex_unlock(&wifi_credentials_mutex);
+	return ret;
+}
+
+int wifi_credentials_store_entry(size_t idx, const void *buf, size_t buf_len)
+{
+	char settings_name_buf[16] = {0};
```

Comment:
We should probably have a comment explaining the size of 16, or make it macro-based

Reply:
I replaced it with `sizeof("wifi_cred/xxx")`, which is smaller than 16, but gets the idea across.

Reply:
I suppose 1000 networks ought to be enough for anybody